core/src/main/java/org/apache/gravitino/authorization/AccessControlManager.java

```diff
@@ -57,44 +61,68 @@ public AccessControlManager(EntityStore store, IdGenerator idGenerator, Config c
   @Override
   public User addUser(String metalake, String user)
       throws UserAlreadyExistsException, NoSuchMetalakeException {
-    return userGroupManager.addUser(metalake, user);
+    return TreeLockUtils.doWithTreeLock(
+        NameIdentifier.of(AuthorizationUtils.ofGroupNamespace(metalake).levels()),
```

Comment:
IIUC, it should be `UserNamespace`, not `GroupNamespace`, right?

Reply:
The current code uses logic:

https://github.com/apache/gravitino/blob/300c79b00578b592cdc152c06c196bfb015b69e6/server/src/main/java/org/apache/gravitino/server/web/rest/UserOperations.java#L69-L88

https://github.com/apache/gravitino/blob/300c79b00578b592cdc152c06c196bfb015b69e6/server/src/main/java/org/apache/gravitino/server/web/rest/UserOperations.java#L120-L143

https://github.com/apache/gravitino/blob/300c79b00578b592cdc152c06c196bfb015b69e6/server/src/main/java/org/apache/gravitino/server/web/rest/UserOperations.java#L145-L169

@jerqi 
Could you please help confirm it? It seems there are some problems with the code.

Reply:
Yes, I used wrong lock. You can correct it.

Reply:
Could you fix the wrong lock of `getUser`?

Reply:
OK, Let me change it.